ibm/resource_ibm_pi_capture.go

```diff
@@ -6,198 +6,221 @@ package ibm
-	//	return err
-	//}
-	return nil
-	//return resourceIBMPIVolumeAttachRead(d, meta)
-
+	d.SetId(fmt.Sprintf("%s/%s", cloudInstanceID, capturename))
+	return resourceIBMPICaptureRead(ctx, d, meta)
 }
 
-func resourceIBMPICaptureRead(d *schema.ResourceData, meta interface{}) error {
-
-	return nil
-}
-
-func resourceIBMPICaptureUpdate(d *schema.ResourceData, meta interface{}) error {
-
-	sess, _ := meta.(ClientSession).IBMPISession()
-	powerinstanceid := d.Get(helpers.PICloudInstanceId).(string)
-	client := st.NewIBMPIVolumeClient(context.Background(), sess, powerinstanceid)
-
-	name := ""
-	if d.HasChange(helpers.PIVolumeAttachName) {
-		name = d.Get(helpers.PIVolumeAttachName).(string)
-	}
-
-	size := float64(d.Get(helpers.PIVolumeSize).(float64))
-	shareable := bool(d.Get(helpers.PIVolumeShareable).(bool))
-
-	volrequest, err := client.UpdateVolume(d.Id(), &models.UpdateVolume{Name: &name, Size: size, Shareable: &shareable})
+func resourceIBMPICaptureRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(ClientSession).IBMPISession()
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
-
-	_, err = isWaitForIBMPIVolumeAvailable(context.Background(), client, *volrequest.VolumeID, d.Timeout(schema.TimeoutCreate))
+	cloudInstanceID, captureID, err := splitID(d.Id())
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
-
-	return resourceIBMPICaptureRead(d, meta)
+	capturedestination := d.Get(helpers.PIInstanceCaptureDestination).(string)
+	if capturedestination != cloudStorageDestination {
+		imageClient := st.NewIBMPIImageClient(ctx, sess, cloudInstanceID)
+		imagedata, err := imageClient.Get(captureID)
+		if err != nil {
+			uErr := errors.Unwrap(err)
+			switch uErr.(type) {
+			case *p_cloud_images.PcloudCloudinstancesImagesGetNotFound:
+				log.Printf("[DEBUG] image does not exist %v", err)
+				d.SetId("")
+				return nil
+			}
+			log.Printf("[DEBUG] get image failed %v", err)
+			return diag.FromErr(err)
+		}
+		imageid := *imagedata.ImageID
+		d.Set("image_id", imageid)
+		d.Set(helpers.PICloudInstanceId, cloudInstanceID)
+	}
+	return nil
 }
 
-func resourceIBMPICaptureDelete(d *schema.ResourceData, meta interface{}) error {
-
-	sess, _ := meta.(ClientSession).IBMPISession()
-	powerinstanceid := d.Get(helpers.PICloudInstanceId).(string)
-	client := st.NewIBMPIVolumeClient(context.Background(), sess, powerinstanceid)
-
-	err := client.DeleteVolume(d.Id())
+func resourceIBMPICaptureDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(ClientSession).IBMPISession()
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
-	// wait for power volume states to be back as available. if it's attached it will be in-use
+	cloudInstanceID, captureID, err := splitID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	capturedestination := d.Get(helpers.PIInstanceCaptureDestination).(string)
```

Comment:
It is better to set the destination in the ID in parts. So that importing resource can work.